wfl/Controllers/RestaurantsController.cs

```diff
@@ -16,27 +13,22 @@ public class RestaurantsController : ApiController
         [Route("")]
         [HttpPost]
         [ResponseType(typeof(Restaurant))]
-        public IHttpActionResult CreateRestaurant([FromUri] string name)
+        public async Task<IHttpActionResult> CreateRestaurantAsync([FromBody] Restaurant restaurant)
         {
-            if (name == null || name.Equals(""))
+            if (restaurant == null || restaurant.Name == null || restaurant.Name.Equals(""))
             {
                 return BadRequest();
             }
-            Restaurant restaurant = new Restaurant()
-            {
-                Name = name,
-                ID = Restaurant.NextID()
-            };
-            Restaurant.Restaurants.Add(restaurant);
-            return Created(name, restaurant);
+            await DBManager.InsertRestaurantAsync(restaurant);
+            return Created(restaurant.ID.ToString(), restaurant);
         }
 
         [Route("")]
         [HttpGet]
-        [ResponseType(typeof(IList<Restaurant>))]
-        public IHttpActionResult Get()
+        [ResponseType(typeof(IEnumerable<Restaurant>))]
```

Comment:
FYI, the `ResponseType` is not required. It is simply metadata that is used by ASP.NET.